Research/lva-node-app/app.js

```diff
@@ -0,0 +1,337 @@
+ */
+
+/**
+ * @requires module:express
+ * @requires module:ws
+ * @requires module:eventhubreader
+ * @requires module:azure-iothub
+ * @requires module:body-parser 
+ * @requires module:express
+ * @requires module:iot-hub-connection-string
+ */
+const express = require('express');
+const WebSocket = require('ws');
+const EventHubReader = require('./eventhubreader.js');
+const iothub = require('azure-iothub');
+const bodyParser = require('body-parser');
+const { request, response } = require('express');
+const { convertIotHubToEventHubsConnectionString } = require('./iot-hub-connection-string.js');
+
+const app = express();
+const server= require('http').createServer(app);
+const wss = new WebSocket.Server({server});
+const Client = iothub.Client;
+
+app.use(bodyParser.urlencoded({extended: false}));
+app.use(bodyParser.json());
+app.use(express.static('public'))
+app.use(express.static('public/pages'))
+
+/**
+ * @global
+ */
+global.DEVICE_ID="";
+global.MODULE_ID="lvaEdge";
+global.IOTHUB_CONNECTION_STRING="";
+global.IOTHUB_ENDPOINT="";
+
+var graphInstances={};
+var graphTopologies={};
+var cameras={};
+var eventHubReader;
+
+const PORT=5000;
+
+/**
+ * For all following declarations of function(req, res):
+ * @param {XMLHttpRequest} req - request object
+ * @param {XMLHttpRequestResponseType} res - request response object
+ */
+
+// render the html pages. the '/<name>' is what the url extension will look like in the browser
+app.get('/', function (req, res) 
+{
+    res.render('index.html')
+})
+app.get('/mediagraph', function (req, res) 
+{
+    res.render('mediagraph.html')
+})
+app.get('/configuration', function (req, res) 
+{
+    res.render('configuration.html')
+})
+app.get('/cameras', function (req, res) 
+{
+    res.render('cameras.html')
+})
+app.get('/output', function (req, res) 
+{
+    res.render('output.html')
+})
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////// 
+// handle GET requests
+
+/**
+* send client global variables upon request 
+* @returns {void}
+*/
+app.get('/getglobals', function (req, res) 
+{
+  console.log("get globals get function!");
+  let globals = 
+  [
+    {
+      "name": "graphInstances",
+      "value": graphInstances
+    },
+    {
+      "name": "graphTopologies",
+      "value": graphTopologies
+    },
+    {
+      "name": "cameras",
+      "value": cameras
+    }
+  ];
+  res.send(JSON.stringify(globals));
+})
+
+/**
+* start sending live stream hub messages to client via websocket 
+* @returns {void}
+*/
+app.get('/hubMessages', function(req, res)
+{
+  if (!validCredentialsAreSet(res))
+  {
+    return;
+  }
+  else
+  {
+    receiveHubMessages();
+    //always close requests
+    res.send();
+  }  
+})
+
+/**
+ * stop receiving messages from iot hub
+ * @returns {void}
+ */
+app.get('/stopMessages', function(req, res)
+{
+  eventHubReader.stopReadMessage();
```

Comment:
if eventHubReader isn't setup (ex we get a stop before a start) this seems dangerous. lets not rely on the UI saving us.

Reply:
the stop read message button is disabled until the user starts receiving messages!

Reply:
lets not rely on the UI saving us.

Reply:
resolved in backend

Reply:
I don't see how this is resolved? If eventHubReader isn't set in app.js this still seems problematic... what is the change in eventHubReader as well that makes this work?